src/Strategy/Puli.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Http\Discovery\Strategy;
+
+use Http\Discovery\Exception\PuliUnavailableException;
+
+/**
+ * @author David de Boer <david@ddeboer.nl>
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+class Puli implements DiscoveryStrategy
+{
+    /**
+     * @var \Puli\GeneratedPuliFactory
+     */
+    private static $puliFactory;
+
+    /**
+     * @var \Puli\Discovery\Api\Discovery
+     */
+    private static $puliDiscovery;
+
+    /**
+     * @return \Puli\GeneratedPuliFactory
+     *
+     * @throws PuliUnavailableException
+     */
+    public static function getPuliFactory()
```

Comment:
why are these methods public? imho private would be enough for them. except the setter.